src/ast/codegen_llvm.cpp

```diff
@@ -1197,20 +1210,42 @@ void CodegenLLVM::visit(FieldAccess &acc)
     else if (field.type.type == Type::string)
     {
       AllocaInst *dst = b_.CreateAllocaBPF(field.type, type.cast_type + "." + acc.field);
-      b_.CreateProbeRead(dst, field.type.size, src);
+      if (type.type == Type::ctx)
+        b_.CREATE_MEMCPY(dst,
+                         b_.CreateIntToPtr(src, field_ty->getPointerTo()),
+                         field.type.size,
+                         1,
+                         true);
```

Comment:
At first, I thought so too, but it turned out that the bpf_map_update_elem only accepts the [pointer in the stack](https://github.com/torvalds/linux/blob/bb6d3fb354c5ee8d6bde2d576eb7220ea09862b9/kernel/bpf/helpers.c#L56) (The definition of `ARG_PTR_TO_MAP_VALUE` is [here](https://github.com/torvalds/linux/blob/bb6d3fb354c5ee8d6bde2d576eb7220ea09862b9/include/linux/bpf.h#L217).) I'll comment on it in the code.

```
% sudo ./src/bpftrace -e 'tracepoint:block:block_rq_issue { @ = args->rwbs; }' -v
BTF: using data from /sys/kernel/btf/vmlinux
Attaching 1 probe...

Error log:
0: (bf) r3 = r1
1: (b7) r1 = 0
2: (7b) *(u64 *)(r10 -8) = r1
last_idx 2 first_idx 0
regs=2 stack=0 before 1: (b7) r1 = 0
3: (18) r1 = 0xffff970c59cab800
5: (07) r3 += 32
6: (bf) r2 = r10
7: (07) r2 += -8
8: (b7) r4 = 0
9: (85) call bpf_map_update_elem#2
R3 type=ctx expected=fp
processed 9 insns (limit 1000000) max_states_per_insn 0 total_states 0 peak_states 0 mark_read 0

Error loading program: tracepoint:block:block_rq_issue
```

(By the way, `printf("%s\n", args->rwbs);` is allowed when doing `expr_ = src` because [it accepts arbitrary memory location](https://github.com/torvalds/linux/blob/bb6d3fb354c5ee8d6bde2d576eb7220ea09862b9/kernel/trace/bpf_trace.c#L618). My guess is because bpf_map_update_elem (and other map functions) can be usable in any bpf program type it accepts only `ARG_PTR_TO_MAP_VALUE`  to prevent kernel data leakage, though as of the tracing BPF program it is pointless since it can read arbitrary memory.)